src/persist-client/src/lib.rs

```diff
@@ -569,6 +592,177 @@ mod tests {
         assert_eq!(listen.read_until(&6).await, expected_events);
     }
 
+    // Appends need to be contiguous for a shard, meaning the lower of an appended batch needs to
+    // match the current shard upper.
```

Comment:
nit: I think "match" here gives the wrong impression. can we say "be <=" or "not be in advance of" or something more precise instead?

Reply:
> nit: I think "match" here gives the wrong impression. can we say "be <=" or "not be in advance of" or something more precise instead?

will do!